event-schema/event_schema.go

```diff
@@ -617,149 +635,184 @@ func getPrivateDataJSON(schemaHash string) []byte {
 	return privateDataJSON
 }
 
-func (manager *EventSchemaManagerT) flushEventSchemas() {
-	var flushDBHandle *sql.DB
-	defer func() {
-		if r := recover(); r != nil {
-			// if the panic happens while flushing events, we need to close the connection
-			if flushDBHandle != nil {
-				_ = flushDBHandle.Close()
-			}
-		}
-	}()
-
+func (manager *EventSchemaManagerT) flushEventSchemas(ctx context.Context) {
 	// This will run forever. If you want to quit in between, change it to ticker and call stop()
 	// Otherwise the ticker won't be GC'ed
 	ticker := time.Tick(flushInterval)
-	for range ticker {
-		if !areEventSchemasPopulated {
-			continue
+	for {
+
+		select {
+		case <-ctx.Done():
+			pkgLogger.Infof("Received signal to stop flushing event schemas returning")
+			return
+
+		case <-ticker:
+			if !areEventSchemasPopulated {
+				continue
+			}
+			if err := manager.flushEventSchemasToDB(ctx); err != nil {
+				pkgLogger.Errorf("Unable to flush event schemas to DB: %v", err)
+			}
+
 		}
 
-		// If needed, copy the maps and release the lock immediately
-		manager.eventModelLock.Lock()
-		manager.schemaVersionLock.Lock()
+	}
+}
+
+// flushEventSchemasToDB is the main function which is responsible
+// for pushing the deltas collected in-memory into the database.
+func (manager *EventSchemaManagerT) flushEventSchemasToDB(ctx context.Context) error {
 
-		schemaVersionsInCache := make([]*SchemaVersionT, 0)
-		for _, sv := range updatedSchemaVersions {
-			schemaVersionsInCache = append(schemaVersionsInCache, sv)
+	// If needed, copy the maps and release the lock immediately
+	manager.eventModelLock.Lock()
+	manager.schemaVersionLock.Lock()
+
+	defer manager.eventModelLock.Unlock()
+	defer manager.schemaVersionLock.Unlock()
+
+	schemaVersionsInCache := make([]*SchemaVersionT, 0)
+	for _, sv := range updatedSchemaVersions {
+		schemaVersionsInCache = append(schemaVersionsInCache, sv)
+	}
+
+	if len(updatedEventModels) == 0 && len(schemaVersionsInCache) == 0 {
+		return nil
+	}
+
+	flushDBHandle := createDBConnection()
+	defer flushDBHandle.Close()
+
+	txn, err := flushDBHandle.Begin()
+	if err != nil {
+		return fmt.Errorf("unable to begin txn to flush event schemas, err: %w", err)
+	}
+
+	if len(updatedEventModels) > 0 {
+		if err := flushEventModels(ctx, txn, updatedEventModels); err != nil {
+			txn.Rollback()
```

Comment:
If we can't rollback, shouldn't we return an error like you do when you can't begin a transaction?